backend/app/services/attachment/chunked_upload.py

```
@@ -0,0 +1,618 @@
+            )
+
+        # Check concurrent upload limit
+        redis_client = self._get_redis_client()
+        user_uploads_key = self._user_uploads_key(user_id)
+        current_uploads = redis_client.scard(user_uploads_key)
+
+        if current_uploads >= MAX_CONCURRENT_UPLOADS:
+            raise ChunkedUploadError(
+                f"Too many concurrent uploads (max: {MAX_CONCURRENT_UPLOADS})",
+                error_code="too_many_uploads",
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)